kerosene/events/mixins/event_mixins.py

```diff
@@ -0,0 +1,125 @@
+from kerosene.events import Event
+from kerosene.training import Status
+from kerosene.training.trainers import Trainer
+
+
+class BatchEventMixin(object):
+    def _on_batch_begin(self):
+        assert isinstance(self, Trainer)
```

Comment:
`assert` here because Pycharm doesn't find which object type is called when firing `self.on_batch_begin()` method.

Reply:
Non, c'est un bug dans pycharm mais l'objet pourrait être quelque chose d'autre qu'un trainer

Reply:
Also, I dont think that mixins is a good package name.. it's like calling a package Classes or Interfaces....does not make sens

Reply:
> Non, c'est un bug dans pycharm mais l'objet pourrait être quelque chose d'autre qu'un trainer

So do we keep the assert or not ? 😛